hotfix / include ptrMatrices in matrix-axis slices

ruairimoran opened the pull request:
Make sure ptrMatrices is included when slicing on matrix axis

Commit:
quick fix for ptrMatrices in slices

alphaville committed:
DTensor slice constructor: set m_d_ptrMatrices

When slicing, m_d_ptrMatrices should be nullptr unless nMats>1

alphaville commented on include/tensor.cuh:
```diff
@@ -659,10 +661,6 @@ DTensor<T>::DTensor(const DTensor<T> &other, size_t axis, size_t from, size_t to
     m_d_data = other.m_d_data + offset;
     m_doDestroyData = false;
     m_doDestroyPtrMatrices = false;
-    if (axis != 2) {
```
I removed this because I put `m_d_ptrMatrices = nullptr;` at the beginning.

alphaville commented on include/tensor.cuh:
```diff
@@ -640,11 +640,13 @@ template<typename T>
         m_numRows = other.m_numRows;
         m_numCols = other.m_numCols;
         m_numMats = len;
+        m_d_ptrMatrices = other.m_d_ptrMatrices + from;
```
This line should be here (inside `if (axis == 2)`)

alphaville committed:
DTensor slice constructor: further testing

alphaville committed:
Final touch (towards v1.3.2)

Get rid of warning in DTensor<T>::createRandomTensor
ci/script: use pushd/popd instead of cd
update CHANGELOG

alphaville reviewed (approved):
@ruairimoran I made a couple of changes - we can merge if you're happy with them.

ruairimoran reviewed (commented):
lgtm
